pipelines/broad/arrays/imputation/Imputation.wdl

```diff
@@ -0,0 +1,357 @@
+          var_in_original = CountChunks.var_in_original,
+          var_in_reference = CountChunks.var_in_reference,
+          bcftools_docker = bcftools_docker_tag
+      }
+
+      if (CheckChunks.valid) {
+
+        call tasks.PrePhaseVariantsEagle {
+          input:
+            dataset_bcf = CheckChunks.valid_chunk_bcf,
+            dataset_bcf_index = CheckChunks.valid_chunk_bcf_index,
+            reference_panel_bcf = referencePanelContig.bcf,
+            reference_panel_bcf_index = referencePanelContig.bcf_index,
+            chrom = referencePanelContig.contig,
+            genetic_map_file = genetic_maps_eagle,
+            eagle_docker = eagle_docker_tag,
+            start = startWithOverlaps,
+            end = endWithOverlaps
+        }
+
+        call tasks.minimac4 {
+          input:
+            ref_panel = referencePanelContig.m3vcf,
+            phased_vcf = PrePhaseVariantsEagle.dataset_prephased_vcf,
+            prefix = "chrom" + "_chunk_" + i +"_imputed",
+            chrom = referencePanelContig.contig,
+            minimac4_docker = minimac4_docker_tag,
+            start = start,
+            end = end,
+            window = chunkOverlaps
+        }
+
+        call tasks.AggregateImputationQCMetrics {
+          input:
+            infoFile = minimac4.info,
+            nSamples = CountSamples.nSamples,
+            basename = output_callset_name + "chrom_" + referencePanelContig.contig + "_chunk_" + i,
+            rtidyverse_docker = rtidyverse_docker_tag
+        }
+
+        call tasks.UpdateHeader {
+          input:
+            vcf = minimac4.vcf,
+            vcf_index = minimac4.vcf_index,
+            ref_dict = ref_dict,
+            basename = "chrom" + "_chunk_" + i +"_imputed",
+            gatk_docker = gatk_docker_tag
+        }
+
+        call tasks.SeparateMultiallelics {
+          input:
+            original_vcf = UpdateHeader.output_vcf,
+            original_vcf_index = UpdateHeader.output_vcf_index,
+            output_basename = "chrom" + referencePanelContig.contig + "_chunk_" + i +"_imputed",
+            bcftools_docker = bcftools_docker_tag
+        }
+
+        call tasks.RemoveSymbolicAlleles {
+          input:
+            original_vcf = SeparateMultiallelics.output_vcf,
+            original_vcf_index = SeparateMultiallelics.output_vcf_index,
+            output_basename = "chrom" + referencePanelContig.contig + "_chunk_" + i +"_imputed",
+            gatk_docker = gatk_docker_tag
+        }
+
+        call tasks.SortIds {
+          input:
+            vcf = RemoveSymbolicAlleles.output_vcf,
+            basename = "chrom" + referencePanelContig.contig + "_chunk_" + i +"_imputed",
+            bcftools_docker = bcftools_docker_tag
+        }
+      }
+    }
+    Array[File] aggregatedImputationMetrics = select_all(AggregateImputationQCMetrics.aggregated_metrics)
+    Array[File] chromosome_vcfs = select_all(SortIds.output_vcf)
```

Comment:
It could be, since it would basically just mean that there aren't enough sites in a particular chunk that overlap the reference panel.  However, the workflow will fail if the number of "invalid" chunks is equal to or larger than the "chunks_fail_threshold" input parameter.  By default we set this to 1, meaning that any invalid chunk will fail the workflow, but there could be a situation where we choose to relax that if we understand why a particular chunk or chunks were being marked invalid.